### PR TITLE
TypeScript: Move base tsconfig to calypso-build

### PR DESCRIPTION
**Pull request by ockham**
#### Changes proposed in this Pull Request

TypeScript: Move base tsconfig to calypso-build. This seems to make sense, since the structure is quite similar to how many of our other configs extend their 'base' versions found in `calypso-build`.

Follow-up from #37632

**Not quite working yet, currently falsely taking stuff in `exclude`d dirs into account.**

Docs: https://www.typescriptlang.org/docs/handbook/tsconfig-json.html

#### Testing instructions

No functional changes.

A project specific typecheck can be run:

```sh
npm run typecheck -- --project client/landing/gutenboarding
```

or for Calypso-the-app:

```sh
npm run typecheck
```

[Check the `typecheck` job](https://circleci.com/gh/Automattic/wp-calypso/521728), it should report the same output as current `master`.
(Same if you run the above commands locally, on this branch vs `master`. Note that Gutenboarding will have one more error on this branch because of some import from `client/`. We'll likely squash those when polishing #37945.)

### CODE REVIEW

**Comment by ockham** [comment]
Should be ready for review!

File: packages/calypso-build/tsconfig.json [hunk 1]
```diff
@@ -14,25 +14,7 @@
...
-		"module": "esnext",
-		"baseUrl": ".."
+		"module": "esnext"
 	},
-	"exclude": [
-		"**/build-module/*",
-		"**/build-style/*",
-		"**/build/*",
-		"**/dist/*",
-		"**/node_modules/*",
```
Comment (sirreal) [inline packages/calypso-build/tsconfig.json]: I wonder if these are general and should be kept?

**Review by sirreal — APPROVED** [review]
Nice, thanks!

File: client/landing/gutenboarding/tsconfig.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-	"extends": "../../../tsconfig/base",
+	"extends": "@automattic/calypso-build/tsconfig",
```
[the remark this answers is not in the document]
Comment (sirreal) [reply client/landing/gutenboarding/tsconfig.json]: Filed a PR to the handbook to clarify: https://github.com/microsoft/TypeScript-Handbook/pull/1261

File: packages/calypso-build/tsconfig.json (hunk 1 above)
Comment (ockham) [reply packages/calypso-build/tsconfig.json]: General enough for `calypso-build` probably! However:

> `files`, `include` and `exclude` from the inheriting config file _overwrite_ those from the base config file.

(https://www.typescriptlang.org/docs/handbook/tsconfig-json.html#configuration-inheritance-with-extends)

So we'll still need to have them in the derived tsconfigs as well.

I'll add them to `calypso-build`'s tsconfig in a follow-up (I'd like to see if I can drop the wildcards).

**Pull request merged**